File: source/downloads/code/LeetCode/Python/Two-Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash_map = dict()
        i = 0
        j = len(nums)-1

        for i in range(len(nums)):
            if (target-nums[i]) in hash_map :
                return [hash_map[target-nums[i]], i]
            hash_map[nums[i]] = i

File: downloads/code/LeetCode/Java/Two-Sum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for (int i=0; i<nums.length;i++){
            if(map.containsKey(target-nums[i])){
                int[] ret = {map.get(target-nums[i]), i};
                return ret;
            }
            map.put(nums[i],i);
        }
        return null;
    }
}
